internal/serverapi/serverapi.go

```diff
@@ -147,6 +147,31 @@ type CreateSnapshotSourceResponse struct {
 	SnapshotStarted bool `json:"snapshotted"` // whether snapshotting has been started
 }
 
+// DeletionResponseSingle contains response of attempted deletion of either snapshot or source.
+type DeletionResponseSingle struct {
+	Type    string `json:"type"`    // did we attempt to delete a snapshot or a source
+	Deleted bool   `json:"deleted"` // whether the object was deleted
+	Error   string `json:"error"`   // any errors
+}
+
+// DeletionResponseBatch contains response of attempted batch deletion.
+type DeletionResponseBatch struct {
+	Items []*DeletionStatus `json:"items"` // status of each item we attempted to delete
+}
+
+// DeletionStatus is used to return information about each item that had deletion attempted.
+type DeletionStatus struct {
+	Type    string `json:"type"`    // was the item we attempted to delete a source or a snapshot
+	ID      string `json:"content"` // ID for the item, this would be a path for a source or a manifest ID for a snapshot
+	Deleted bool   `json:"deleted"` // was the item deleted
+	Error   string `json:"error"`   // any errors
+}
+
+// BatchSnapshotDeleteRequest contains a list of snapshot IDs to be deleted.
+type BatchSnapshotDeleteRequest struct {
+	Snapshots []string `json:"snapshots"`
```

Comment:
Can we rename this to `snapshotManifestIds` ? The notion of snapshot ID is slightly ambiguous.

Reply:
yep, have made the change.